File: JAVA/Basics/prime.java
package Basics;
public class prime {
    // public static void main(String[] args) {
    // int n = 15;
    // int cnt= 0;
    // for(int i = 1; i*i <= n ; i ++){

    // if(n%i==0){
    // cnt++;
    // if((n/i)!=i){
    // cnt++;
    // }
    // }
    // }if (cnt == 2){
    // System.out.println(n+" is prime number");
    // }else{
    // System.out.println(n+ " is composite number");
    // }

    // }

    public static void main(String[] args) {
        int n1 = 20;
        int n2 = 18;
        while (n1 > 0 && n2 > 0) {
            if (n1 > n2) {
                n1 = n1 % n2;
            }
            else{
                n2 = n2 % n1;
            }
        }
        System.out.println((n1 == 0) ? n2 : n1);
    }
}

File: PYTHON/Basics/prime.py
import math
n = 7
cnt = 0
for i in range (1, int(math.sqrt(n))+1):
    if n % i ==0:
        cnt+=1
        if (n//i)!=i:
            cnt +=1
if cnt ==2:
    print("this is the prime")
else:
    print("this is not a prime")

    
a = 25
b = 10

while a > 0 and b > 0:
    if a > b:
        a = a % b
    else: 
        b = b % a
print(b if a == 0 else a)



